packages/amplify-graphql-default-value-transformer/src/__tests__/amplify-grapphql-default-value-transformer.test.ts

```diff
@@ -338,4 +198,90 @@ describe('DefaultValueModelTransformer:', () => {
     expect(out.resolvers['Mutation.createNote.init.1.req.vtl']).toBeDefined();
     expect(out.resolvers['Mutation.createNote.init.2.req.vtl']).toBeUndefined();
   });
+
+  it.each([{ strategy: mockSqlDataSourceStrategy() }])(
+    'throws if auto-increment is implied on a non-Postgres datasource',
+    ({ strategy }) => {
+      const schema = `
+      type CoffeeQueue @model {
+        id: ID! @primaryKey
+        orderNumber: Int! @default
+        name: String
+      }`;
+      expect(() => {
+        testTransform({
+          schema: schema,
+          transformers: [new ModelTransformer(), new DefaultValueTransformer(), new PrimaryKeyTransformer()],
+          dataSourceStrategies: constructDataSourceStrategies(schema, strategy),
+        });
+      }).toThrow('The @default directive requires a value property on non Postgres datasources.');
+    },
+  );
+
+  it.each([
+    { typeStr: 'Boolean' },
+    { typeStr: 'AWSJSON' },
+    { typeStr: 'AWSDate' },
+    { typeStr: 'AWSDateTime' },
+    { typeStr: 'AWSTime' },
+    { typeStr: 'AWSTime' },
+    { typeStr: 'AWSURL' },
+    { typeStr: 'AWSPhone' },
+    { typeStr: 'AWSIPAddress' },
+  ])('throws if auto-increment is implied on non-int types', ({ typeStr }) => {
+    const strategy = mockSqlDataSourceStrategy({ dbType: POSTGRES_DB_TYPE });
+
+    expect(() => {
+      const schema = `
+      type Test @model {
+        id: ID! @primaryKey
+        value: ${typeStr} @default
+      }
+    `;
+      testTransform({
+        schema,
+        transformers: [new ModelTransformer(), new DefaultValueTransformer(), new PrimaryKeyTransformer()],
+        dataSourceStrategies: constructDataSourceStrategies(schema, strategy),
+      });
+    }).toThrow('The empty @default (auto-increment) may only be applied to integer fields.');
+  });
+
+  it('should successfully transform a schema that implies auto-increment', async () => {
```

Comment:
What does "implies auto-increment" mean?

Reply:
the only defined use of `@default` is for auto-increment fields but that is not explicit in the word `@default`. That being said I agree that my wording is clunky. Re-written.